pkg/agent/agent.go

```diff
@@ -818,6 +831,34 @@ func (i *Initializer) initializeIPSec() error {
 	return nil
 }
 
+// initializeWireguard checks if preconditions are met for using WireGuard and initialize WireGuard client or clean up.
+func (i *Initializer) initializeWireGuard() error {
+
+	mtu := i.nodeConfig.NodeLocalInterfaceMTU - config.WireGuardOverhead
+	// create and initialize WireGuard client
+	v4Enabled := config.IsIPv4Enabled(i.nodeConfig, i.networkConfig.TrafficEncapMode)
+	v6Enabled := config.IsIPv6Enabled(i.nodeConfig, i.networkConfig.TrafficEncapMode)
+
+	var nodeIPv4, nodeIPv6 net.IP
+	nodeIPv4 = i.nodeConfig.NodeIPAddr.IP.To4()
+
+	if nodeIPv4 == nil {
+		nodeIPv6 = i.nodeConfig.NodeIPAddr.IP.To16()
+	}
+
+	wgClient, err := wireguard.New(i.nodeConfig.Name, i.client, mtu, nodeIPv4, nodeIPv6,
+		v4Enabled, v6Enabled, i.networkConfig.WireGuardPort)
+	if err != nil {
+		return err
+	}
+	i.wireGuardClient = wgClient
+	// clean up routes
+	if i.networkConfig.TrafficEncryptionMode != config.TrafficEncryptionModeWireGuard {
+		return i.wireGuardClient.Cleanup()
+	}
+	return i.wireGuardClient.Init()
```

Comment:
Do we have a way to check WF kernel module is loaded?

Reply:
wireGuardClient.Init() will return an error indicating the kernel module is missing if the error code is `EOPNOTSUPP` when setting up the WireGuard interface.

Reply:
Ok